pkg/kubelet/lifecycle/predicate.go

```diff
@@ -227,7 +229,7 @@ func GeneralPredicates(pod *v1.Pod, nodeInfo *schedulerframework.NodeInfo) ([]Pr
 	}
 
 	var reasons []PredicateFailureReason
-	for _, r := range noderesources.Fits(pod, nodeInfo) {
+	for _, r := range noderesources.Fits(pod, nodeInfo, utilfeature.DefaultFeatureGate.Enabled(features.PodOverhead)) {
```

Comment:
Removing the features dep from the plugin means this is the only way to pass that value in to `Fits`, so we need to make the change with it

Reply:
right, I am just suggesting to do it in a follow up PR so we get sig-node to review a significantly smaller one.

Reply:
btw, I think this `Fit` function is only used by kubelet

Reply:
I understand, but my point is if we don't react to the change in this PR then any kubelet/lifecycle tests that rely on this feature gate will fail

Reply:
And not updating the `Fits` signature means we need to somehow pass a podOverhead value (hardcoded?) to `computePodResourceRequest`. That's the function shared by all the scorers. I think it is just reasonable to get sig-node to be aware of this change now

Reply:
While unfortunate that we can't isolate kubelet changes, I agree with Mike.

Reply:
Can't we update Fits as follows:
```
// Fits checks if node have enough resources to host the pod.
func Fits(pod *v1.Pod, nodeInfo *framework.NodeInfo) []InsufficientResource {
	return fitsRequest(computePodResourceRequest(pod, utilfeature.DefaultFeatureGate.Enabled(features.PodOverhead)), nodeInfo, nil, nil)
}
```

Reply:
@ahg-g that would keep the featuregate dep in the plugin package. We could do that now, and then actually remove the dep in a follow up if you'd like

Reply:
yes, removing the dependency is basically a change to the build file and removing the import. It is up to you, but this is my advise if you want to make faster progress :)

Reply:
Perhaps just splitting the change into a separate commit will ease the work for the kubelet reviewers, it is such a minor change that I think it will be quicker to get it done here if that's all right